openedx/stanford/common/djangoapps/enrollment/views.py

```diff
@@ -51,54 +51,59 @@ def post(self, request):
             }
         """
         try:
-            course_id = CourseKey.from_string(request.data['course_id'])
+            course_id = request.data['course_id']
+            course_key = CourseKey.from_string(course_id)
         except InvalidKeyError:
             return Response(
                 status=status.HTTP_400_BAD_REQUEST,
-                data={"message": u"Invalid or missing course_id"},
+                data={
+                    'message': u'Invalid or missing course_id',
+                },
             )
-
-        if not user_has_role(request.user, CourseStaffRole(course_id)):
+        if not user_has_role(request.user, CourseStaffRole(course_key)):
             return Response(
                 status=status.HTTP_403_FORBIDDEN,
-                data={"message": u"User does not have permission to update enrollment for [{course_id}].".format(course_id=course_id)},
+                data={
+                    'message': u'User does not have permission to update enrollment for [{course_id}].'.format(
+                        course_id=course_id,
+                    ),
+                },
             )
-
         email = request.data['email']
         try:
             validate_email(email)
         except ValidationError:
             return Response(
                 status=status.HTTP_400_BAD_REQUEST,
-                data={"message": u"Invalid email address"},
+                data={
+                    'message': u'Invalid email address',
+                },
             )
-
         action = request.data['action']
         email_students = request.POST.get('email_students', False) in ['true', 'True', True]
-        auto_enroll = request.POST.get('email_students', False) in ['true', 'True', True]
-
+        auto_enroll = request.POST.get('auto_enroll', False) in ['true', 'True', True]
```

Comment:
@kluo Do we actually care to be this forgiving? Shouldn't we just enforce one style?

Reply:
Well I was maintaining how it currently checks from instructor dashboard:
https://github.com/Stanford-Online/edx-platform/blob/master/lms/djangoapps/instructor/views/api.py#L3349